staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -313,10 +314,15 @@ func DefaultKubernetesUserAgent() string {
 // running inside a pod running on kubernetes. It will return ErrNotInCluster
 // if called from a process not running in a kubernetes environment.
 func InClusterConfig() (*Config, error) {
-	const (
-		tokenFile  = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	tokenFile := ""
+	rootCAFile := ""
+	if gruntime.GOOS == "windows" {
+		tokenFile = filepath.Join(homedir.HomeDir(), "/var/run/secrets/kubernetes.io/serviceaccount/token")
```

Comment:
Please follow the pattern shown here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1alpha3/defaults_windows.go#L23

you can add a windows specific file with either a constant or a function that prefixes using `homedir.HomeDir` and another file for other platforms which does not use that.

Reply:
basically avoid the `gruntime.GOOS == "windows"` check

Reply:
OK, thanks!

Reply:
@dims I am sorry and I do not understand what your means. I may add one windows file, but here I still need to check whether the OS is 'windows'. If it is windows and then assign to this variable.
I am looking forward to your reply.

Reply:
In defaults.go
```
defaultTokenFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
defaultRootCAFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
```

In defaults_windows.go
```

import (
  "path/filepath"
  "k8s.io/client-go/util/homedir"
)

defaultTokenFile = filepath.Join(homedir.HomeDir(), "/var/run/secrets/kubernetes.io/serviceaccount/token")
defaultRootCAFile = filepath.Join(homedir.HomeDir(), "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")

```

Then you can just use `defaultTokenFile` and `defaultRootCAFile` and won't need the `gruntime.GOOS == "windows"` condition